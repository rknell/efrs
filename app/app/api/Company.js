import call from './call'

class FormsApi {
  addGroup (group) {
    return call('/company/group', 'POST', group)
  }

  deleteGroup (groupId) {
    return call(`/company/group/${groupId}`, 'DELETE')
  }

  getStyle (domain) {
    return call(`/company/styles/${domain}`, 'GET', undefined, true)
  }

  create (data) {
    return call(`/company/create`, 'POST', data)
  }
}

export default new FormsApi()
