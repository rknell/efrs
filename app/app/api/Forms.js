class FormsApi {

  // save (form) {
  //   return call(`/form`, 'POST', form)
  // }

  // delete (formId) {
  //   return call(`/form/${formId}`, 'DELETE')
  // }
}

export default new FormsApi()
