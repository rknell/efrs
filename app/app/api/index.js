import user from './User'
import forms from './Forms'
import responses from './Response'
import company from './Company'
import settings from './Settings'
import call from './call'

export default {
  user, forms, responses, company, settings, _call: call
}
