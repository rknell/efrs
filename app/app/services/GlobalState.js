import { observable, observe, spy } from 'mobx'
import { Alert, AsyncStorage, Linking } from 'react-native'
import PushNotifications from './PushNotifications'
import Translate from '../services/Translate'

class GlobalState {
  // apiPath = 'http://192.168.1.8:1676/api'
  apiPath = 'https://www.efrs.io/api'
  // apiPath = 'http://localhost:1676/api'

  @observable model = {
    userToken: null,
    user: null,
    userPermissions: {
      permissions: {},
      groups: []
    },
    canPop: false,
    forms: [],
    cachedCalls: {
      get: {},
      post: []
    },
    isBusy: 0,
    pushToken: null
  }

  current = () => {
    return {
      user: this.model.user ? this.model.user._id : 'no user',
      group: this.model.userPermissions.groups.join(',')
    }
  }

  navigation = {
    resetTo: () => {
      console.log('RESETTING NAVIGATION')
    },
    push: () => {
      console.log('PUSHING NAVIGATION')
    }
  }

  saveSetting (key, value) {
    return AsyncStorage.setItem(`@EFRS.AppSettings:${key}`, value)
  }

  getSetting (key) {
    return AsyncStorage.getItem(`@EFRS.AppSettings:${key}`)
  }

  constructor () {
    spy(event => {
      if (event.type === 'action' || event.type === 'update') {
        // console.log(event);
      }
    })

    this.getSetting('formsData').then(data => {
      if (data) this.model.forms = JSON.parse(data)
    })

    const PNObj = new PushNotifications(this._setPushToken)
    this.pushNotifications = PNObj.messaging

    observe(this.model, 'user', () => {
      try {
        if (this.model.user && this.bugsnag) {
          this.bugsnag.setUser(this.model.user._id, `${this.model.user.firstName} ${this.model.user.lastName}`, this.model.user.email)
        }
      } catch (e) {
        console.error('could not set user for bugsnag', e)
      }
    })
  }

  _setPushToken = (token) => {
    this.model.pushToken = token
  }

  openUrl (url) {
    Linking.canOpenURL(url).then(supported => {
      if (!supported) {
        Alert.alert(Translate.do('Error'), Translate.do('Can\'t open this url on this device. Try installing an app that can handle this link type and try again.'))
      } else {
        return Linking.openURL(url)
      }
    }).catch(err => console.error('An error occurred', err))
  }
}

export default new GlobalState()
