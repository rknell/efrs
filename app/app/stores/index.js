import React from 'react'
import NetworkRequestHandler from './NetworkRequestHandler'
import GlobalState from '../services/GlobalState'
import { Alert, AsyncStorage } from 'react-native'
import Home from '../pages/Home'
import Domain from '../pages/Domain'
import UsersStore from './UsersStore'
import GroupsStore from './GroupsStore'
import ReportStore from './ReportStore'
import ResponseStore from './ResponseStore'
import FormStore from './FormStore'
import AssignedToGroupStore from './AssignedToGroupStore'
import AssignedToUserStore from './AssignedToUserStore'
import { computed } from 'mobx'
import TranslationStore from './TranslationStore'
import Translate from '../services/Translate'
import ScheduledResponsesStore from './ScheduledResponses'

class Stores {
  API_PATH = 'https://www.efrs.io/api'
  // API_PATH = 'http://192.168.1.8:1676/api'
  // API_PATH = 'http://localhost:1676/api'

  constructor () {
    this.errorHandler = this.errorHandler.bind(this)
    this.networkRequestHandler = new NetworkRequestHandler(this.API_PATH, false, this.errorHandler)

    this.groups = new GroupsStore(this.networkRequestHandler)
    this.reports = new ReportStore(this.networkRequestHandler)
    this.responses = new ResponseStore(this.networkRequestHandler, this)
    this.forms = new FormStore(this.networkRequestHandler)
    this.assignedToGroup = new AssignedToGroupStore(this.networkRequestHandler)
    this.assignedToUser = new AssignedToUserStore(this.networkRequestHandler)
    this.users = new UsersStore(this.networkRequestHandler)
    this.translations = new TranslationStore(this.networkRequestHandler)
    this.scheduledResponses = new ScheduledResponsesStore(this.networkRequestHandler)

    this.stores = [
      this.groups, this.reports, this.responses, this.forms, this.assignedToGroup, this.assignedToUser, this.users, this.scheduledResponses
    ]
    this.initialise()
  }

  initialise = async () => {
    const userToken = await this.loadNetworkCredentials()

    if (userToken) {
      this.refreshAll()
    }
  }

  refreshAll () {
    if (this.networkRequestHandler.headers['Authorization']) {
      this.stores.forEach(store => store.refresh())
    }
  }

  clearAll = async () => {
    for (let store of this.stores) {
      await store.clearCached()
    }
  }

  async errorHandler (error = {}) {
    if (error.status === 401) {
      const domain = await AsyncStorage.getItem(`@login:domain`)
      if (domain) {
        // User has logged in before, go to home
        GlobalState.navigation.resetTo({ view: <Home /> })
      } else {
        GlobalState.navigation.resetTo({ view: <Domain /> })
      }
    } else if (error.body) {
      // Using a timeout because mobx has a problem with sync with the system alert.
      if (error.body.message) {
        Alert.alert(Translate.do('Error'), Translate.do(error.body.message))
      } else {
        Alert.alert(Translate.do('Error'), Translate.do('An unknown server error occurred'))
      }
      throw (error)
    } else {
      throw (error)
    }
  }

  loadNetworkCredentials = async () => {
    const userToken = await AsyncStorage.getItem('@PASettings:authToken')

    if (userToken) {
      this.networkRequestHandler.setHeader('Authorization', `bearer ${userToken}`)
    }
    return userToken
  }

  @computed get outbox () {
    return [
      ...this.forms.data.filter(item => item.isDirty),
      ...this.responses.data.filter(item => item.isDirty)
    ]
  }

  outboxSync = async () => {
    for (let item of this.outbox) {
      await item.sync()
    }
  }
}

export default new Stores()
