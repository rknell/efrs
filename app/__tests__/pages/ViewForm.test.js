import React from 'react'
// Note: test renderer must be required after react-native.
import ViewFormView, { ExportView, ViewFormController } from '../../app/pages/ViewForm'

import formFixtures from '../../../server/tests/fixtures/forms'
import responseFixtures from '../../../server/tests/fixtures/responses'
import basicForm from '../../../server/tests/fixtures/forms/basicForm'
import formLib from '../../app/lib/form'
import { shallow } from 'enzyme'
import { Alert, AsyncStorage, Linking, Platform } from 'react-native'
import GlobalState from '../../app/services/GlobalState'
import api from '../../app/api'
import stores from '../../app/stores'

import toJSON from 'enzyme-to-json'

AsyncStorage.setItem = (key, value) => {
  console.log('ASYNC-SET', key, value)
  return Promise.resolve()
}

AsyncStorage.getItem = (key) => {
  console.log('ASYNC-GET', key)
  if (key === `efrs.drafts.TEST`) {
    return Promise.resolve('[]')
  } else {
    return Promise.resolve('{test: true}')
  }
}

describe('ViewFormView', () => {
  // No snapshot because data is mutated when opened
  it('renders without error', () => {
    const copyForm = JSON.parse(JSON.stringify(formFixtures[0]))
    formLib.hydrateValues(copyForm, responseFixtures[0])
    shallow(<ViewFormView form={copyForm} />)
  })

  it('renders android without error', () => {
    const copyForm = JSON.parse(JSON.stringify(formFixtures[0]))
    formLib.hydrateValues(copyForm, responseFixtures[0])
    Platform.OS = 'android'
    shallow(<ViewFormView form={copyForm} />)
  })
})

describe('ExportView', () => {
  it('renders an exportView', () => {
    const copyForm = JSON.parse(JSON.stringify(formFixtures[0]))
    const controller = new ViewFormController({ form: copyForm })
    const wrapper = shallow(<ExportView controller={controller} />)
    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('renders an exportView with destination email', () => {
    const copyForm = JSON.parse(JSON.stringify(formFixtures[0]))
    const controller = new ViewFormController({ form: copyForm })
    controller.model.exportSettings.destination = 'email'
    const wrapper = shallow(<ExportView controller={controller} />)
    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('sets export destination', () => {
    const copyForm = JSON.parse(JSON.stringify(formFixtures[0]))
    const controller = new ViewFormController({ form: copyForm })
    const instance = shallow(<ExportView controller={controller} />).instance()
    instance.setExportDestination('TEST')
    expect(controller.model.exportSettings.destination).toBe('TEST')
  })

  it('sets export view', () => {
    const copyForm = JSON.parse(JSON.stringify(formFixtures[0]))
    const controller = new ViewFormController({ form: copyForm })
    const instance = shallow(<ExportView controller={controller} />).instance()
    instance.setExportView('TEST')
    expect(controller.model.exportSettings.view).toBe('TEST')
  })

  it('sets export view', () => {
    const copyForm = JSON.parse(JSON.stringify(formFixtures[0]))
    const controller = new ViewFormController({ form: copyForm })
    const instance = shallow(<ExportView controller={controller} />).instance()
    instance.setExportFormat('TEST')
    expect(controller.model.exportSettings.format).toBe('TEST')
  })
})

describe('FormController', () => {
// eslint-disable-next-line no-undef
  beforeEach(() => {
    if (!stores.forms) {
      stores.forms = {
        data: []
      }
    }
    stores.forms.data.length = 0
    formFixtures.forEach(form => {
      stores.forms.data.push(form)
    })
    stores.forms.data.push(basicForm)
  })

  it('should instantiate with a response', () => {
    const response = { ...responseFixtures[0] }
    const form = { ...formFixtures[0] }
    response.formId = form._id
    let gotSettings = false
    GlobalState.getSetting = (setting) => {
      gotSettings = true
      return Promise.resolve('TEST')
    }
    const controller = new ViewFormController({ form, response })
    expect(gotSettings).toBe(true)
    expect(controller).toBeDefined()
  })

  it('should handle onGoBack, no changes', () => {
    const response = { ...responseFixtures[0] }
    const form = { ...formFixtures[0] }
    const controller = new ViewFormController({ form, response })
    let didPop = false
    GlobalState.navigation.pop = () => {
      didPop = true
    }
    controller.onGoBack()
    expect(didPop).toBe(true)
  })

  it('should handle onGoBack, changes, dont save', () => {
    const response = { ...responseFixtures[0] }
    const form = { ...formFixtures[0] }
    const controller = new ViewFormController({ form, response })
    controller.formHasChanged = () => true
    let didPop = false
    GlobalState.navigation.pop = () => {
      didPop = true
    }

    Alert.alert = (title, message, options) => {
      expect(options[0].text).toBeDefined()
      options[0].onPress()
    }
    controller.onGoBack()
    expect(didPop).toBe(true)
  })

  it('should handle onGoBack, changes, save', () => {
    const response = { ...responseFixtures[0] }
    const form = { ...formFixtures[0] }
    const controller = new ViewFormController({ form, response })
    controller.formHasChanged = () => true

    // let didPop = false
    // GlobalState.navigation.pop = () => {
    //   didPop = true
    // }

    Alert.alert = (title, message, options) => {
      Alert.alert = jest.fn()
      console.log('Options log', options, title, message)
      expect(options[1].text).toBeDefined()
      options[1].onPress()
    }

    controller.onGoBack()
    // expect(didPop).toBe(true)
  })

  it('should save a draft', done => {
    const response = { ...responseFixtures[0] }
    const form = { ...formFixtures[0] }
    const controller = new ViewFormController({ form, response })

    let didPop = false
    GlobalState.navigation.pop = () => {
      didPop = true
    }
    GlobalState.model.user = { _id: 'TEST' }
    controller.draft()
      .then(() => {
        expect(didPop).toBe(true)
        done()
      })
      .catch(done)
  })

  it('should save when valid', async () => {
    const response = { ...responseFixtures[0], formId: basicForm._id }
    const form = { ...basicForm }
    let didOnPop = false

    const controller = new ViewFormController({
      form,
      response,
      onPop: () => {
        didOnPop = true
      }
    })

    let didPop = false
    GlobalState.navigation.pop = () => {
      didPop = true
    }

    controller.formController = { unassignAll: () => {} }
    await controller.save()
    expect(didPop).toBe(true)
    expect(didOnPop).toBe(true)
  })

  it('should save when valid, no onPop', () => {
    const response = { ...responseFixtures[0], formId: basicForm._id }
    const form = { ...basicForm }
    let didOnPop = false
    const controller = new ViewFormController({
      form, response
    })

    let didPop = false
    GlobalState.navigation.pop = () => {
      didPop = true
    }
    controller.formController = { unassignAll: () => {} }
    return controller.save()
      .then(() => {
        expect(didPop).toBe(true)
        expect(didOnPop).toBe(false)
      })
  })

  it('should save, dont go back', () => {
    const response = { ...responseFixtures[0], formId: basicForm._id }
    const form = { ...basicForm }
    let didOnPop = false
    const controller = new ViewFormController({
      form,
      response,
      onPop: () => {
        didOnPop = true
      }
    })

    let didPop = false
    GlobalState.navigation.pop = () => {
      didPop = true
    }

    controller.formController = { unassignAll: () => {} }
    controller.model.originalForm = { ...basicForm }
    controller.model.form = { ...basicForm, isTest: true }

    return controller.save(true)
      .then(() => {
        expect(didPop).toBe(false)
        expect(didOnPop).toBe(false)
        expect(controller.model.originalForm).toEqual(JSON.stringify({ ...basicForm, isTest: true, isIncomplete: false }))
      })
  })

  it('should toggle email input', () => {
    const response = { ...responseFixtures[0], formId: basicForm._id }
    const form = { ...basicForm }
    const controller = new ViewFormController({ form, response })

    controller.toggleEmailInput()
    controller.toggleEmailInput()
  })

  describe('email', () => {
    it('should email', () => {
      const response = { ...responseFixtures[0], formId: basicForm._id }
      const form = { ...basicForm }
      const controller = new ViewFormController({ form, response })
      controller.email()
    })

    it('should email, and toggle input', () => {
      const response = { ...responseFixtures[0], formId: basicForm._id }
      const form = { ...basicForm }
      const controller = new ViewFormController({ form, response })
      controller.email({ dontToggleEmail: false })
    })

    it('should email, and save changes', done => {
      const response = { ...responseFixtures[0], formId: basicForm._id }
      const form = { ...basicForm }
      const controller = new ViewFormController({ form, response })
      controller.formController = { unassignAll: () => {} }
      controller.formHasChanged = () => true
      Alert.alert = (title, message, options) => {
        Alert.alert = jest.fn()
        expect(title).toBeDefined()
        expect(message).toBeDefined()
        options[1].onPress()
          .then(() => done())
          .catch(done)
      }
      controller.email({ dontToggleEmail: false })
    })

    it('should email, dont toggle email', done => {
      const response = { ...responseFixtures[0], formId: basicForm._id }
      const form = { ...basicForm }
      const controller = new ViewFormController({ form, response })

      const oldEmail = api.responses.email
      api.responses.email = (email, responseId) => {
        api.responses.email = oldEmail
        return Promise.resolve()
      }

      controller.email({ dontToggleEmail: true })
        .then(() => {
          done()
        })
        .catch(done)
    })

    it('should email, no message', done => {
      const response = { ...responseFixtures[0], formId: basicForm._id }
      const form = { ...basicForm }
      const controller = new ViewFormController({ form, response })

      const oldEmail = api.responses.email
      api.responses.email = (email, responseId) => {
        api.responses.email = oldEmail
        return Promise.resolve({})
      }

      controller.email({ dontToggleEmail: true })
        .then(() => {
          done()
        })
        .catch(done)
    })

    it('should email, throw error', done => {
      const response = { ...responseFixtures[0], formId: basicForm._id }
      const form = { ...basicForm }
      const controller = new ViewFormController({ form, response })

      const oldEmail = api.responses.email
      api.responses.email = (email, responseId) => {
        api.responses.email = oldEmail
        return Promise.reject(new Error('TEST ERROR'))
      }

      controller.email({ dontToggleEmail: true })
        .then(() => {
          done()
        })
        .catch(done)
    })
  })

  describe('emailCSV', () => {
    it('should emailCSV, and toggle input', () => {
      const response = { ...responseFixtures[0], formId: basicForm._id }
      const form = { ...basicForm }
      const controller = new ViewFormController({ form, response })
      controller.emailCSV({ dontToggleEmail: false })
    })

    it('should emailCSV, and save changes', done => {
      const response = { ...responseFixtures[0], formId: basicForm._id }
      const form = { ...basicForm }
      const controller = new ViewFormController({ form, response })
      controller.formController = { unassignAll: () => {} }
      controller.formHasChanged = () => true
      Alert.alert = (title, message, options) => {
        Alert.alert = jest.fn()
        expect(title).toBeDefined()
        expect(message).toBeDefined()
        options[1].onPress()
          .then(() => done())
          .catch(done)
      }
      controller.emailCSV({ dontToggleEmail: false })
    })

    it('should emailCSV, dont toggle email', done => {
      const response = { ...responseFixtures[0], formId: basicForm._id }
      const form = { ...basicForm }
      const controller = new ViewFormController({ form, response })

      const oldEmail = api.responses.emailCSV
      api.responses.emailCSV = (email, responseId) => {
        api.responses.emailCSV = oldEmail
        return Promise.resolve()
      }

      controller.emailCSV({ dontToggleEmail: true })
        .then(() => {
          done()
        })
        .catch(done)
    })

    it('should emailCSV, no message', done => {
      const response = { ...responseFixtures[0], formId: basicForm._id }
      const form = { ...basicForm }
      const controller = new ViewFormController({ form, response })

      const oldEmail = api.responses.emailCSV
      api.responses.emailCSV = (email, responseId) => {
        api.responses.emailCSV = oldEmail
        return Promise.resolve({})
      }

      controller.emailCSV({ dontToggleEmail: true })
        .then(() => {
          done()
        })
        .catch(done)
    })

    it('should emailCSV, throw error', done => {
      const response = { ...responseFixtures[0], formId: basicForm._id }
      const form = { ...basicForm }
      const controller = new ViewFormController({ form, response })

      const oldEmail = api.responses.emailCSV
      api.responses.emailCSV = (email, responseId) => {
        api.responses.emailCSV = oldEmail
        return Promise.reject(new Error('TEST ERROR'))
      }

      controller.emailCSV({ dontToggleEmail: true })
        .then(() => {
          done()
        })
        .catch(done)
    })
  })

  describe('view', () => {
    it('should open link', () => {
      const response = { ...responseFixtures[0], formId: basicForm._id }
      const form = { ...basicForm }
      const controller = new ViewFormController({ form, response })
      const oldLink = Linking.openURL

      let didOpenUrl = false
      Linking.openURL = (url) => {
        Linking.openURL = oldLink
        didOpenUrl = true
      }

      controller.formController = { unassignAll: () => {} }
      controller.view()

      expect(didOpenUrl).toBe(true)
    })

    it('should open link, has changed', done => {
      const response = { ...responseFixtures[0], formId: basicForm._id }
      const form = { ...basicForm }
      const controller = new ViewFormController({ form, response })
      const oldLink = Linking.openURL

      controller.formHasChanged = () => true
      controller.formController = { unassignAll: () => {} }

      let didOpenUrl = false
      Linking.openURL = (url) => {
        Linking.openURL = oldLink
        didOpenUrl = true
      }

      Alert.alert = (title, message, options) => {
        Alert.alert = jest.fn()
        controller.formHasChanged = () => false
        options[1].onPress().then(() => {
          expect(didOpenUrl).toBe(true)
          done()
        }).catch(done)
      }

      controller.view()
    })
  })

  describe('viewCSV', () => {
    it('should open link', () => {
      const response = { ...responseFixtures[0], formId: basicForm._id }
      const form = { ...basicForm }
      const controller = new ViewFormController({ form, response })
      const oldLink = Linking.openURL

      let didOpenUrl = false
      Linking.openURL = (url) => {
        Linking.openURL = oldLink
        didOpenUrl = true
      }

      controller.formController = { unassignAll: () => {} }
      controller.viewCSV()

      expect(didOpenUrl).toBe(true)
    })

    it('should open link, has changed', done => {
      const response = { ...responseFixtures[0], formId: basicForm._id }
      const form = { ...basicForm }
      const controller = new ViewFormController({ form, response })
      const oldLink = Linking.openURL

      controller.formHasChanged = () => true
      controller.formController = { unassignAll: () => {} }

      let didOpenUrl = false
      Linking.openURL = (url) => {
        Linking.openURL = oldLink
        didOpenUrl = true
      }

      Alert.alert = (title, message, options) => {
        Alert.alert = jest.fn()
        controller.formHasChanged = () => false
        options[1].onPress().then(() => {
          expect(didOpenUrl).toBe(true)
          done()
        }).catch(done)
      }

      controller.viewCSV()
    })
  })
})
