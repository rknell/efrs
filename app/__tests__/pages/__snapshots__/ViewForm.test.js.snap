// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExportView renders an exportView 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <NavBar
    title="Export"
  />
  <QuickPick
    globalStyle={
      Object {
        "disabledText": "rgb(179, 179, 179)",
        "errorColour": "red",
        "fullLogo": Object {
          "testUri": "../../../app/assets/efrsFullBanner.png",
        },
        "neutralColour": "white",
        "neutralColourDisabled": "rgba(255, 255, 255, 0.3)",
        "neutralColourHighlight": "rgb(217, 217, 217)",
        "neutralColourIsDark": false,
        "neutralColourOffset": "rgb(242, 242, 242)",
        "neutralColourText": "rgb(51, 51, 51)",
        "neutralColourTextDisabled": "rgba(51, 51, 51, 0.3)",
        "primaryColour": "dodgerblue",
        "primaryColourDisabled": "rgba(30, 144, 255, 0.3)",
        "primaryColourHighlight": "rgb(106, 182, 255)",
        "primaryColourIsDark": true,
        "primaryColourText": "rgb(255, 255, 255)",
        "primaryColourTextDisabled": "rgba(255, 255, 255, 0.3)",
        "secondaryColour": "#f3f3f3",
        "secondaryColourDisabled": "rgba(243, 243, 243, 0.3)",
        "secondaryColourHighlight": "rgb(205, 205, 205)",
        "secondaryColourIsDark": undefined,
        "secondaryColourText": "rgb(39, 39, 39)",
        "secondaryColourTextDisabled": "rgba(39, 39, 39, 0.3)",
        "secondaryLogo": Object {
          "testUri": "../../../app/assets/icon.png",
        },
      }
    }
    onChange={[Function]}
    options={
      Array [
        Object {
          "label": "Report",
          "value": "report",
        },
      ]
    }
    settings={
      Object {
        "Label": "Export",
      }
    }
    value="report"
  />
  <QuickPick
    globalStyle={
      Object {
        "disabledText": "rgb(179, 179, 179)",
        "errorColour": "red",
        "fullLogo": Object {
          "testUri": "../../../app/assets/efrsFullBanner.png",
        },
        "neutralColour": "white",
        "neutralColourDisabled": "rgba(255, 255, 255, 0.3)",
        "neutralColourHighlight": "rgb(217, 217, 217)",
        "neutralColourIsDark": false,
        "neutralColourOffset": "rgb(242, 242, 242)",
        "neutralColourText": "rgb(51, 51, 51)",
        "neutralColourTextDisabled": "rgba(51, 51, 51, 0.3)",
        "primaryColour": "dodgerblue",
        "primaryColourDisabled": "rgba(30, 144, 255, 0.3)",
        "primaryColourHighlight": "rgb(106, 182, 255)",
        "primaryColourIsDark": true,
        "primaryColourText": "rgb(255, 255, 255)",
        "primaryColourTextDisabled": "rgba(255, 255, 255, 0.3)",
        "secondaryColour": "#f3f3f3",
        "secondaryColourDisabled": "rgba(243, 243, 243, 0.3)",
        "secondaryColourHighlight": "rgb(205, 205, 205)",
        "secondaryColourIsDark": undefined,
        "secondaryColourText": "rgb(39, 39, 39)",
        "secondaryColourTextDisabled": "rgba(39, 39, 39, 0.3)",
        "secondaryLogo": Object {
          "testUri": "../../../app/assets/icon.png",
        },
      }
    }
    onChange={[Function]}
    options={
      Array [
        Object {
          "label": "CSV",
          "value": "csv",
        },
        Object {
          "label": "PDF",
          "value": "html",
        },
      ]
    }
    settings={
      Object {
        "Label": "To Format",
      }
    }
    value={null}
  />
  <QuickPick
    globalStyle={
      Object {
        "disabledText": "rgb(179, 179, 179)",
        "errorColour": "red",
        "fullLogo": Object {
          "testUri": "../../../app/assets/efrsFullBanner.png",
        },
        "neutralColour": "white",
        "neutralColourDisabled": "rgba(255, 255, 255, 0.3)",
        "neutralColourHighlight": "rgb(217, 217, 217)",
        "neutralColourIsDark": false,
        "neutralColourOffset": "rgb(242, 242, 242)",
        "neutralColourText": "rgb(51, 51, 51)",
        "neutralColourTextDisabled": "rgba(51, 51, 51, 0.3)",
        "primaryColour": "dodgerblue",
        "primaryColourDisabled": "rgba(30, 144, 255, 0.3)",
        "primaryColourHighlight": "rgb(106, 182, 255)",
        "primaryColourIsDark": true,
        "primaryColourText": "rgb(255, 255, 255)",
        "primaryColourTextDisabled": "rgba(255, 255, 255, 0.3)",
        "secondaryColour": "#f3f3f3",
        "secondaryColourDisabled": "rgba(243, 243, 243, 0.3)",
        "secondaryColourHighlight": "rgb(205, 205, 205)",
        "secondaryColourIsDark": undefined,
        "secondaryColourText": "rgb(39, 39, 39)",
        "secondaryColourTextDisabled": "rgba(39, 39, 39, 0.3)",
        "secondaryLogo": Object {
          "testUri": "../../../app/assets/icon.png",
        },
      }
    }
    onChange={[Function]}
    options={
      Array [
        Object {
          "label": "Download",
          "value": "download",
        },
        Object {
          "label": "Email",
          "value": "email",
        },
      ]
    }
    settings={
      Object {
        "Label": "Destination",
      }
    }
    value={null}
  />
  <Button
    color="dodgerblue"
    onPress={[Function]}
    title="Export"
  />
</View>
`;

exports[`ExportView renders an exportView with destination email 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <NavBar
    title="Export"
  />
  <QuickPick
    globalStyle={
      Object {
        "disabledText": "rgb(179, 179, 179)",
        "errorColour": "red",
        "fullLogo": Object {
          "testUri": "../../../app/assets/efrsFullBanner.png",
        },
        "neutralColour": "white",
        "neutralColourDisabled": "rgba(255, 255, 255, 0.3)",
        "neutralColourHighlight": "rgb(217, 217, 217)",
        "neutralColourIsDark": false,
        "neutralColourOffset": "rgb(242, 242, 242)",
        "neutralColourText": "rgb(51, 51, 51)",
        "neutralColourTextDisabled": "rgba(51, 51, 51, 0.3)",
        "primaryColour": "dodgerblue",
        "primaryColourDisabled": "rgba(30, 144, 255, 0.3)",
        "primaryColourHighlight": "rgb(106, 182, 255)",
        "primaryColourIsDark": true,
        "primaryColourText": "rgb(255, 255, 255)",
        "primaryColourTextDisabled": "rgba(255, 255, 255, 0.3)",
        "secondaryColour": "#f3f3f3",
        "secondaryColourDisabled": "rgba(243, 243, 243, 0.3)",
        "secondaryColourHighlight": "rgb(205, 205, 205)",
        "secondaryColourIsDark": undefined,
        "secondaryColourText": "rgb(39, 39, 39)",
        "secondaryColourTextDisabled": "rgba(39, 39, 39, 0.3)",
        "secondaryLogo": Object {
          "testUri": "../../../app/assets/icon.png",
        },
      }
    }
    onChange={[Function]}
    options={
      Array [
        Object {
          "label": "Report",
          "value": "report",
        },
      ]
    }
    settings={
      Object {
        "Label": "Export",
      }
    }
    value="report"
  />
  <QuickPick
    globalStyle={
      Object {
        "disabledText": "rgb(179, 179, 179)",
        "errorColour": "red",
        "fullLogo": Object {
          "testUri": "../../../app/assets/efrsFullBanner.png",
        },
        "neutralColour": "white",
        "neutralColourDisabled": "rgba(255, 255, 255, 0.3)",
        "neutralColourHighlight": "rgb(217, 217, 217)",
        "neutralColourIsDark": false,
        "neutralColourOffset": "rgb(242, 242, 242)",
        "neutralColourText": "rgb(51, 51, 51)",
        "neutralColourTextDisabled": "rgba(51, 51, 51, 0.3)",
        "primaryColour": "dodgerblue",
        "primaryColourDisabled": "rgba(30, 144, 255, 0.3)",
        "primaryColourHighlight": "rgb(106, 182, 255)",
        "primaryColourIsDark": true,
        "primaryColourText": "rgb(255, 255, 255)",
        "primaryColourTextDisabled": "rgba(255, 255, 255, 0.3)",
        "secondaryColour": "#f3f3f3",
        "secondaryColourDisabled": "rgba(243, 243, 243, 0.3)",
        "secondaryColourHighlight": "rgb(205, 205, 205)",
        "secondaryColourIsDark": undefined,
        "secondaryColourText": "rgb(39, 39, 39)",
        "secondaryColourTextDisabled": "rgba(39, 39, 39, 0.3)",
        "secondaryLogo": Object {
          "testUri": "../../../app/assets/icon.png",
        },
      }
    }
    onChange={[Function]}
    options={
      Array [
        Object {
          "label": "CSV",
          "value": "csv",
        },
        Object {
          "label": "PDF",
          "value": "html",
        },
      ]
    }
    settings={
      Object {
        "Label": "To Format",
      }
    }
    value={null}
  />
  <QuickPick
    globalStyle={
      Object {
        "disabledText": "rgb(179, 179, 179)",
        "errorColour": "red",
        "fullLogo": Object {
          "testUri": "../../../app/assets/efrsFullBanner.png",
        },
        "neutralColour": "white",
        "neutralColourDisabled": "rgba(255, 255, 255, 0.3)",
        "neutralColourHighlight": "rgb(217, 217, 217)",
        "neutralColourIsDark": false,
        "neutralColourOffset": "rgb(242, 242, 242)",
        "neutralColourText": "rgb(51, 51, 51)",
        "neutralColourTextDisabled": "rgba(51, 51, 51, 0.3)",
        "primaryColour": "dodgerblue",
        "primaryColourDisabled": "rgba(30, 144, 255, 0.3)",
        "primaryColourHighlight": "rgb(106, 182, 255)",
        "primaryColourIsDark": true,
        "primaryColourText": "rgb(255, 255, 255)",
        "primaryColourTextDisabled": "rgba(255, 255, 255, 0.3)",
        "secondaryColour": "#f3f3f3",
        "secondaryColourDisabled": "rgba(243, 243, 243, 0.3)",
        "secondaryColourHighlight": "rgb(205, 205, 205)",
        "secondaryColourIsDark": undefined,
        "secondaryColourText": "rgb(39, 39, 39)",
        "secondaryColourTextDisabled": "rgba(39, 39, 39, 0.3)",
        "secondaryLogo": Object {
          "testUri": "../../../app/assets/icon.png",
        },
      }
    }
    onChange={[Function]}
    options={
      Array [
        Object {
          "label": "Download",
          "value": "download",
        },
        Object {
          "label": "Email",
          "value": "email",
        },
      ]
    }
    settings={
      Object {
        "Label": "Destination",
      }
    }
    value="email"
  />
  <TextFieldView
    globalStyle={
      Object {
        "disabledText": "rgb(179, 179, 179)",
        "errorColour": "red",
        "fullLogo": Object {
          "testUri": "../../../app/assets/efrsFullBanner.png",
        },
        "neutralColour": "white",
        "neutralColourDisabled": "rgba(255, 255, 255, 0.3)",
        "neutralColourHighlight": "rgb(217, 217, 217)",
        "neutralColourIsDark": false,
        "neutralColourOffset": "rgb(242, 242, 242)",
        "neutralColourText": "rgb(51, 51, 51)",
        "neutralColourTextDisabled": "rgba(51, 51, 51, 0.3)",
        "primaryColour": "dodgerblue",
        "primaryColourDisabled": "rgba(30, 144, 255, 0.3)",
        "primaryColourHighlight": "rgb(106, 182, 255)",
        "primaryColourIsDark": true,
        "primaryColourText": "rgb(255, 255, 255)",
        "primaryColourTextDisabled": "rgba(255, 255, 255, 0.3)",
        "secondaryColour": "#f3f3f3",
        "secondaryColourDisabled": "rgba(243, 243, 243, 0.3)",
        "secondaryColourHighlight": "rgb(205, 205, 205)",
        "secondaryColourIsDark": undefined,
        "secondaryColourText": "rgb(39, 39, 39)",
        "secondaryColourTextDisabled": "rgba(39, 39, 39, 0.3)",
        "secondaryLogo": Object {
          "testUri": "../../../app/assets/icon.png",
        },
      }
    }
    onChange={[Function]}
    settings={
      Object {
        "Auto Capitalize": "none",
        "Auto Correct": false,
        "Keyboard Type": "email-address",
        "Label": "Email",
      }
    }
    value=""
  />
  <Button
    color="dodgerblue"
    onPress={[Function]}
    title="Export"
  />
</View>
`;
